bfasst/transform/spydrnet_injector.py

```diff
@@ -0,0 +1,199 @@
+"""Tool to inject errors into a netlist"""
+from random import randrange, sample
+import spydrnet as sdn
+from bfasst.transform.base import TransformTool
+from bfasst.status import Status, TransformStatus
+
+
+class SpydrNetErrorInjector(TransformTool):
+    """Tool to inject errors into a netlist"""
+
+    success_status = Status(TransformStatus.SUCCESS)
+    TOOL_WORK_DIR = "spydrnet_injector"
+
+    def __init__(self, cwd, design) -> None:
+        super().__init__(cwd, design)
+        self.clean_netlist = sdn.parse(self.design.reversed_netlist_path)
+        self.hierarchical_luts = []
+        self.flattened_luts = None
+
+    def inject(self, error_type):
+        """Injects an error into the netlist of the given type"""
+        if error_type == "BIT_FLIP":
+            self.inject_bit_flip()
+        elif error_type == "WIRE_SWAP":
+            self.inject_wire_swap()
+        else:
+            return Status(TransformStatus.ERROR, "Invalid error type")
+        return Status(TransformStatus.SUCCESS)
+
+    def inject_bit_flip(self):
+        """Injects a bit flip error into the netlist"""
+        num_luts = self.pick_luts_from_netlist()
+        self.flatten_luts()
+        lut_number = randrange(num_luts)
+        lut_size = self.get_lut_size(lut_number)
+        bit_number = randrange(lut_size)
+        self.flip_bit(lut_number, bit_number)
+        self.compose_corrupt_netlist()
+
+    def pick_luts_from_netlist(self):
+        """Calculates the number of LUTs in the netlist"""
+        num_luts = 0
+        for library in self.clean_netlist.libraries:
+            for definition in library.definitions:
+                if "LUT" in definition.name.upper():
+                    num_luts += len(definition.references)
+                    self.hierarchical_luts.append(definition.references)
+        return num_luts
+
+    def flatten_luts(self):
```

Comment:
"flatten" has other meanings for netlists.  Let's not use that term here.  Just `get_all_luts` is fine.

Reply:
Done.